src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingStackTraceService.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.API
+{
+    internal sealed class UnitTestingStackTraceService
+    {
+        private readonly Workspace _workspace;
+        private readonly IStackTraceExplorerService _stackTraceService;
+
+        public UnitTestingStackTraceService(Workspace workspace)
+        {
+            _workspace = workspace;
+            _stackTraceService = workspace.Services.GetRequiredService<IStackTraceExplorerService>();
+        }
+
+        public async Task<ImmutableArray<Frame>> TryParseAsync(string input, CancellationToken cancellationToken)
+        {
+            var result = await StackTraceAnalyzer.AnalyzeAsync(input, cancellationToken).ConfigureAwait(false);
+            return result.ParsedFrames.SelectAsArray(p => new Frame(p, _stackTraceService, _workspace));
+        }
+
+        public readonly struct Frame
```

Comment:
📝 Every type in the external access layer should have the target prefix so Navigate To doesn't get annoying. For data wrapper, the name should match the original name but have a `Wrapper` suffix.

```suggestion
        public readonly struct UnitTestingParsedFrameWrapper
```